specs/altair/sync-protocol.md

```diff
@@ -256,3 +351,368 @@ def process_light_client_update(store: LightClientStore,
+
+`GetLightClientBootstrap` is primarily used by light clients to obtain the initial `BeaconBlockHeader` and the corresponding sync committee from a checkpoint block root.
+
+The request MUST be encoded as an SSZ-field.
+
+Clients MUST be able to produce `LightClientBootstrap` instances for roots of epoch boundary blocks seen on the epoch range `[max(GENESIS_EPOCH, current_epoch - MIN_EPOCHS_FOR_BLOCK_REQUESTS), current_epoch]` where `current_epoch` is defined by the current wall-clock time. Clients MAY also provide `LightClientBootstrap` instances for non-epoch boundary blocks.
+
+To form a `LightClientBootstrap`, the requested block's post state
+`trusted_state` is needed.
+
+```python
+def create_light_client_bootstrap(trusted_state: BeaconState) -> LightClientBootstrap:
+    return LightClientBootstrap(
+        header=BeaconBlockHeader(
+            slot=trusted_state.latest_block_header.slot,
+            proposer_index=trusted_state.latest_block_header.proposer_index,
+            parent_root=trusted_state.latest_block_header.parent_root,
+            state_root=trusted_state.hash_tree_root(),
+            body_root=trusted_state.latest_block_header.body_root,
+        ),
+        current_sync_committee=trusted_state.current_sync_committee,
+        current_sync_committee_branch=build_proof(trusted_state.get_backing(), CURRENT_SYNC_COMMITTEE_INDEX)
+    )
+```
+
+## Honest Validator
+
+This section extends the [honest validator specification for Altair](./validator.md) with additional responsibilities to enable light clients to sync with the network.
+
+### Beacon chain responsibilities
+
+When assigned block proposer duties, validators have a new responsibility
+to broadcast a corresponding `OptimisticLightClientUpdate` to the network.
+
+#### Block proposal
+
+After broadcasting a block with a `sync_aggregate` fulfilling
+`sum(sync_aggregate.sync_committee_bits) >= MIN_SYNC_COMMITTEE_PARTICIPANTS`
+validators SHOULD also produce a corresponding `OptimisticLightClientUpdate`
+and send it on the `optimistic_light_client_update` pubsub topic
+after one-third of the next slot has transpired
+(`SECONDS_PER_SLOT / 3` seconds after the start of the next slot).
+
+To form an `OptimisticLightClientUpdate`, the following are needed:
+- `state`: latest state (produced block's post state)
+- `block`: `state`'s parent block (the produced block)
+- `attested_state`: `block`'s parent state (produced block's pre state)
+
+```python
+def create_optimistic_light_client_update(state: BeaconState,
+                                          block: SignedBeaconBlock,
+                                          attested_state: BeaconState) -> OptimisticLightClientUpdate:
+    slot = block.message.slot
+    assert slot == state.slot
+    attested_header = attested_state.latest_block_header.copy()
+    attested_header.state_root = attested_state.hash_tree_root()
+    assert state.latest_block_header.parent_root == attested_header.hash_tree_root()
```

Comment:
it may be nice to check `assert state.latest_block_header.body_root == block.message.body.hash_tree_root()` to further validate `sync_aggregate`

Reply:
Thanks, I have extended the validation to the full `BeaconBlockHeader`.